### PR TITLE
SPT-4536: Bugfix for array type generation

### PR DESCRIPTION
**Pull request by jondavidford**
In `SchemaGenerator`, we were checking `if (processedDefinitions.Contains(property.Id))` before the check for `if (property.Type != null && property.Type.Value.HasFlag(JSchemaType.Array) && schema.Properties.Add(keyValuePair.Key, new SchemaObject {Ref = $"#/definitions/{property.Id}"}); processedDefinitions.Contains(property.Items.First().Id))`, which meant in the case a model contained multiple lists of the same type, a corrupt reference like `#/definitions/System.Collections.Generic.List1[[Core.Models.Workflow.Stage, Core, Version=9.0.0.0, Culture=neutral, PublicKeyToken=null]]` was getting added to the schema rather than the correct `array` type.

### CODE REVIEW

File: src/QSwagGenerator/Generators/SchemaGenerator.cs [hunk 1]
```diff
@@ -109,11 +109,17 @@ internal SchemaObject MapToSchema(JSchema jSchema, HashSet<Uri> processedDefinit
                 foreach (var keyValuePair in jSchema.Properties)
```
Comment (dmikov) [inline src/QSwagGenerator/Generators/SchemaGenerator.cs]: Use tuple decomposition

File: src/QSwagGenerator/Generators/SchemaGenerator.cs [hunk 2]
```diff
@@ -109,11 +109,17 @@ internal SchemaObject MapToSchema(JSchema jSchema, HashSet<Uri> processedDefinit
...
                 {
                     var property = keyValuePair.Value;
-                    if (processedDefinitions.Contains(property.Id))
-                        schema.Properties.Add(keyValuePair.Key, new SchemaObject {Ref = $"#/definitions/{property.Id}"});
-                    else if (property.Type != null && property.Type.Value.HasFlag(JSchemaType.Array) && processedDefinitions.Contains(property.Items.First().Id))
-                        schema.Properties.Add(keyValuePair.Key, new SchemaObject{Type=SchemaType.Array,
-                                Items = new List<SchemaObject> {new SchemaObject { Ref = $"#/definitions/{property.Items.First().Id}" } } });
+                    if (property.Type != null && property.Type.Value.HasFlag(JSchemaType.Array) &&
```
Comment (dmikov) [inline src/QSwagGenerator/Generators/SchemaGenerator.cs]: Use null propagation?

File: src/QSwagGenerator/Generators/SchemaGenerator.cs [hunk 3]
```diff
@@ -109,11 +109,17 @@ internal SchemaObject MapToSchema(JSchema jSchema, HashSet<Uri> processedDefinit
...
+                        processedDefinitions.Contains(property.Items.First().Id))
```
Comment (dmikov) [inline src/QSwagGenerator/Generators/SchemaGenerator.cs]: What if collection is empty? Re-write in reverse. `property.Items.Any(item=>processDefiniton.Contains(item.Id))`

**Comment by jondavidford** [comment]
I will make these changes, but all I did was switch the position of these blocks in the if statement. I assumed it was already in the desired style.

**Pull request merged**